### PR TITLE
Helpers update

### PR DESCRIPTION
**Pull request by huv1k**
Improves `API` helpers to lazy load data and a possibility to opt-out from body parsing.

### CODE REVIEW

**Review by lfades — APPROVED** [review]
I like this usage of get/set 💯

File: packages/next-server/server/api-utils.ts [hunk 1]
```diff
@@ -56,13 +59,34 @@ function parseJson(str: string) {
  * @returns Object with key name of query argument and its value
  */
 export function parseQuery({ url }: IncomingMessage) {
```
Comment (lfades) [inline packages/next-server/server/api-utils.ts]: probably better to name it `getQueryParser`, to match it with `getCookieParser`

Comment (huv1k) [reply packages/next-server/server/api-utils.ts]: Could you make a suggestion please? 🙏

**Review by ijjk — APPROVED** [review]

File: packages/next-server/server/next-server.ts [hunk 2]
```diff
@@ -313,18 +316,28 @@ export default class Server {
     }
 
     try {
+      const resolverModule = require(resolverFunction)
+
+      if (resolverModule.config) {
+        const config: IPageConfig = resolverModule.config
+        if (config.api && !config.api.bodyParser) {
```
Comment (ijjk) [inline packages/next-server/server/next-server.ts]:
```suggestion
        if (config.api && config.api.bodyParser === false) {
```

File: packages/next/types/index.d.ts [hunk 3]
```diff
@@ -48,6 +48,9 @@ export type NextPage<P = {}, IP = P> = {
  */
 export type PageConfig = {
   amp?: boolean | 'hybrid'
+  api?: {
+    bodyParser?: boolean
```
Comment (Janpot) [inline packages/next/types/index.d.ts]: The problem with having a static export is that it always has to be configure by the user. Third party libraries can't configure the handlers they produce to opt out of this body-parser. e.g. if I create a proxy library, I'd have to ask my users to always include this static export:
```js
import someProxy from 'third-party';
export default someProxy();
// I'd have to ask users of my library to include the following or it won't work:
export const {
  api: {
    bodyParser: false
  }
}
```

**Pull request merged**